test/java/seedu/addressbook/util/TestUtil.java

```diff
@@ -75,4 +80,19 @@ public static void assertTextFilesEqual(Path path1, Path path2) throws IOExcepti
         List<String> list2 = Files.readAllLines(path2, Charset.defaultCharset());
         assertEquals(String.join("\n", list1), String.join("\n", list2));
     }
+
+    /**
+     * Executes the command, and asserts the result message is as expected.
+     */
+    public static void assertCommandResult(Command command, AddressBook addressBook,
+                                           AddressBook expectedAddressBook, String expectedMessage) {
```

Comment:
Problems with this method:

1. Naming: `CommandResult` is a class, but `assertCommandResult` does more than check for `CommandResult`. Maybe this should be called `assertCommandBehavior` instead?

2. Not watertight: it assumes that the passed-in `addressBook` is the address book that has been registered with the command. Which may not be true, and opens the door to false negatives.

3. It does not check for the `getRelevantPersons()` of the returned `CommandResult`.